openhcl/openhcl_boot/src/host_params/dma_hint.rs

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Calculate DMA hint value if not provided by host.
+
+use crate::log;
+use igvm_defs::{MemoryMapEntryType, PAGE_SIZE_4K};
+use super::PartitionInfo;
+
+struct DmaLookupStruct {
+    /// Logical processors for VM.
+    vp_count: u32,
+    /// Vtl2AddressRangeSize - Vtl2MmioAddressRangeSize.
+    vtl2_memory_mb: u32,
+    /// DMA hint in MiB.
+    dma_hint_mb: u32,
+}
+
+/// Lookup table for DMA hint calculation.
+const LOOKUP_TABLE: &'static [DmaLookupStruct] = &[
+    DmaLookupStruct {
+        vp_count: 2,
+        vtl2_memory_mb: 98,
+        dma_hint_mb: 4,
+    },
+    DmaLookupStruct {
+        vp_count: 4,
+        vtl2_memory_mb: 110,
+        dma_hint_mb: 6,
+    },
+    DmaLookupStruct {
+        vp_count: 8,
+        vtl2_memory_mb: 148,
+        dma_hint_mb: 10,
+    },
+    DmaLookupStruct {
+        vp_count: 16,
+        vtl2_memory_mb: 256,
+        dma_hint_mb: 18,
+    },
+    DmaLookupStruct {
+        vp_count: 32,
+        vtl2_memory_mb: 516,
+        dma_hint_mb: 36,
+    },
+    DmaLookupStruct {
+        vp_count: 48,
+        vtl2_memory_mb: 718,
+        dma_hint_mb: 52,
+    },
+    DmaLookupStruct {
+        vp_count: 64,
+        vtl2_memory_mb: 924,
+        dma_hint_mb: 68,
+    },
+    DmaLookupStruct {
+        vp_count: 96,
+        vtl2_memory_mb: 1340,
+        dma_hint_mb: 102,
+    },
+];
+
+/// Returns calculated DMA hint value, in 4k pages.
```

Comment:
I'd probably want some more rationale on _what_ we should do if we don't match one of these lookup tables exactly.